notebooks/MAST/K2/K2_Lightcurve/k2_lightcurve.ipynb

```diff
@@ -0,0 +1,334 @@
+    "\n",
+    "Now that we have the light curve file, let's store the timestamps and fluxes as arrays for use later."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with fits.open(fits_file, mode=\"readonly\") as hdulist:\n",
+    "    k2_bjds = hdulist[1].data['TIME']\n",
+    "    sap_fluxes = hdulist[1].data['SAP_FLUX']\n",
+    "    pdcsap_fluxes = hdulist[1].data['PDCSAP_FLUX']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plot the light curve.\n",
+    "\n",
+    "Let's make a plot of the PDCSAP flux vs. time in Kepler BJD."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
+    "\n",
+    "# Plot the timeseries in black circles.\n",
+    "ax.plot(k2_bjds, pdcsap_fluxes, 'ko')\n",
+    "\n",
+    "# Let's label the axes and define a title for the figure.\n",
+    "fig.suptitle(\"TRAPPIST-1 Light Curve - Campaign 12\")\n",
+    "ax.set_ylabel(\"PDCSAP Flux (e-/s)\")\n",
+    "ax.set_xlabel(\"Time (BKJD)\")\n",
+    "\n",
+    "# Let's zoom in on the x-axis and y-axis.  We can see a sinusoidal pattern due to starspots.\n",
+    "# The transits are in there too, but you'd need to clean the light curve before you see them.\n",
+    "ax.set_xlim(2920., 2950.)\n",
+    "ax.set_ylim(0.1e7, 0.2e7)\n",
+    "\n",
+    "# Adjust the left margin so the y-axis label shows up.\n",
+    "plt.subplots_adjust(left=0.15)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Understanding Light Curve Flags\n",
+    "\n",
+    "The table of information contains more than just timestamps and fluxes.  In particular, there is a column of flags associated with every timestamp that indicate a number of warnings and conditions associated with that measurement.  Not every flag is worth excluding from your analysis: you should always make your own decision.  A summary of the flags can be found [here](https://archive.stsci.edu/kepler/manuals/archive_manual.pdf#page=19)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First we need to read in the array of cadence quality flags, let's do\n",
+    "# that now.\n",
+    "with fits.open(fits_file, mode=\"readonly\") as hdulist:\n",
+    "    qual_flags = hdulist[1].data['SAP_QUALITY']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's plot the full time series, but this time we'll overplot those points that have a quality flag greater than zero in red."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Start figure and axis.\n",
+    "fig, ax = plt.subplots()\n",
```

Comment:
Suggesting making this plot larger too (`figsize=(12,8)`?)

Reply:
Size adjusted.